src/app/repositories/firebase/dataset_firebase_repository.py

```diff
@@ -92,6 +152,26 @@ def get_unit_supplementary_data(
             .to_dict()
         )
 
+    def get_number_of_unit_supplementary_data_with_dataset_id(
+        self, dataset_id: str
+    ) -> int:
+        """
+        Get the number of unit supplementary data associated with a dataset id.
+
+        Parameters:
+        dataset_id (str): The unique id of the dataset
+        """
+
+        returned_unit_data = (
+            self.datasets_collection.document(dataset_id).collection("units").stream()
+        )
+
+        unit_data_count = 0
+        for unit_data in returned_unit_data:
```

Comment:
unit_data is unused

Reply:
I have switched to count unit_data with cursor as the old way cause 530 query timed out error